variables.tf

```diff
@@ -83,9 +83,8 @@ variable "enable_lifecycle_configuration_rules" {
 variable "lifecycle_configuration_rules" {
   type = list(object({
     id      = string
-    prefix  = string
     enabled = bool
-    tags    = map(string)
+    filter  = any
```

Comment:
can we define type of filter also?

Reply:
We can set it to `filter  = object({})` but then this will show changes after every `terraform apply` as below -
```bash
Note: Objects have changed outside of Terraform

Terraform detected the following changes made outside of Terraform since the last "terraform apply" which may have
affected this plan:

  # module.test_s3_bucket.aws_s3_bucket_lifecycle_configuration.default[0] has changed
  ~ resource "aws_s3_bucket_lifecycle_configuration" "default" {
        id     = "test-s3-bucket"
        # (1 unchanged attribute hidden)

      ~ rule {
            id     = "test-s3-bucket-30days"
            # (1 unchanged attribute hidden)

          ~ filter {
              + prefix = "myfolder1/myfolder2/"

              - and {
                  - object_size_greater_than = 0 -> null
                  - object_size_less_than    = 0 -> null
                  - prefix                   = "myfolder1/myfolder2/" -> null
                }
            }

            # (1 unchanged block hidden)
        }
    }


Unless you have made equivalent changes to your configuration, or ignored the relevant attributes using ignore_changes,
the following plan may include actions to undo or respond to these changes.

────────────────────────────────────────────────────────────────────────────────────────────────────────────────────────
<--- terraform changes --->
Plan: 0 to add, 1 to change, 0 to destroy.
```